packages/math-input/src/components/input/math-input.tsx

```diff
@@ -12,19 +12,18 @@ import {
     wonderBlocksBlue,
     offBlack,
 } from "../common-style";
-import LegacyKeypad from "../keypad-legacy";
 
 import CursorHandle from "./cursor-handle";
 import DragListener from "./drag-listener";
 import MathWrapper from "./math-wrapper";
 import {scrollIntoView} from "./scroll-into-view";
 
-import type {Cursor} from "../../types";
+import type {Cursor, KeypadAPI} from "../../types";
 
 const constrainingFrictionFactor = 0.8;
 
 type Props = {
-    keypadElement: LegacyKeypad;
+    keypadElement?: KeypadAPI;
```

Comment:
If someone were to go directly to an exercise URL, the keypad and the renderer (I believe) would be rendering at the same time. So technically there's no guarantee that the keypad would exist before the input exists?